scripts/test.sh

```diff
@@ -97,33 +99,31 @@ for i in "${ignored_packages[@]}"; do
   serial_packages=(${serial_packages[@]//*$i*})
 done
 
-if [ "${1:-""}" = "" ]; then
+if [[ -z "${restrict_testing_to}" ]]; then
```

Comment:
[Medium] I think that having an if else statement in here leads to code duplication because the things done in the else statement are almost the same as the ones within the if statement with the only difference that we need to check if they are subdirectories of the `restrict_testing_to` directory.

With this said I think that we could have 2 separate functions. One for running the tests for the serial packages and another for running the tests of the non-serial packages.

In fact it could be even turned into a single function but let's first extract those 2.

These functions should both accept a `restrict_testing_to` variable but with the difference that when we're are not restricting to any directory we set `restrict_testing_to` to be the root directory, i.e. `/`.

Reply:
Cleaned this up a bit while also making this whole section more understandable